### ISSUE TITLE
CFE should give more detail when failing on "dart:js" import

### ISSUE THREAD

**Issue by davidmorgan** [dart-lang/sdk#34065]
Before the switch to dart 2/CFE, an attempt to import 'dart:js' from VM code would tell you which file imports 'dart:js':

```
Unable to spawn isolate: The built-in library 'dart:js' is not available on the stand-alone VM.
'foo/bar.dart': error: line 1 pos 1: library handler failed
import 'dart:js';
```

Now we are missing this information:

```
Failed to load "/path/to/main.dart": Unable to spawn isolate: dart:js: Error: Not found: dart:js.
```

making it hard to figure out where the `dart:js` import came from.

**Comment by mraleph** [comment]
@kmillikin for triage.

**Comment by kmillikin** [comment]
Be assured that it has some attention.  It's on our (unfortunately long) list of error messages that aren't informative enough.  Fixing those generally falls at a lower priority than implementing missing or incorrect Dart 2 features.

Without more context it's hard to see elevating the priority of this any higher than it already is.

**Comment by dskloetg** [comment]
This is a regression. Please reconsider your priorities. I don't think it's reasonable to break stuff by adding features while down-prioritizing fixing the things you break in order to add even more features (that in turn might break more things).

**Comment by mraleph** [comment]
This is severe usability issue - the error is "not informative *at all*", rather than "not informative *enough*".

**Comment by kmillikin** [comment]
We really do understand that it is a severe usability issue.

It's not completely simple for us to handle it because we detect the problem too late, when we try to compile dart:js not when we see the import of dart:js.

I have a change that will make it possible for us to fix this issue.  ETA is early next week but I'll try to update here if that slips.

**Comment by matanlurey** [comment]
Any update?

I had to revert a CL I spent 2-3 days on because I couldn't track down where an import came from.

**Comment by dgrove** [comment]
Any updates, @kmillikin ?

**Comment by kmillikin** [comment]
This CL fixes the problem: https://dart-review.googlesource.com/c/sdk/+/71161

It changes the representation of empty programs and I need to make sure that's OK with the VM.

Before:

```
dart:js:1: Error: Not found: dart:js.
```

After:

```
file:///[...snip...]/kmillikin/programs/dart/a.dart:1:1: Error: Not found: 'dart:js'
import 'dart:js';
^
```

**Comment by matanlurey** [comment]
Closed by @kmillikin in https://dart-review.googlesource.com/c/sdk/+/71161. Thanks!

**Comment by dskloetg** [comment]
When will this be available in google3?

**Comment by kmillikin** [comment]
My fix from above had to be reverted because it didn't work with hot reload.  The real fix was landed in the Dart SDK with 56c2ed5e9ea8791274d0deda083a1b4e077a512f on September 13.
